src/frontend/src/monitor/stats.rs

```diff
@@ -80,3 +84,124 @@ impl FrontendMetrics {
+            "subscription_cursor_error_count",
+            "The subscription error num of cursor",
+            registry
+        )
+        .unwrap();
+        let opts = histogram_opts!(
+            "subscription_cursor_query_duration",
+            "The amount of time a query exists inside the cursor",
+            exponential_buckets(1.0, 5.0, 11).unwrap(),
+        );
+        let subscription_cursor_query_duration =
+            register_histogram_vec_with_registry!(opts, &["subscription_name"], registry).unwrap();
+
+        let opts = histogram_opts!(
+            "subscription_cursor_declare_duration",
+            "Subscription cursor duration of declare",
+            exponential_buckets(1.0, 5.0, 11).unwrap(),
+        );
+        let subscription_cursor_declare_duration =
+            register_histogram_vec_with_registry!(opts, &["subscription_name"], registry).unwrap();
+
+        let opts = histogram_opts!(
+            "subscription_cursor_fetch_duration",
+            "Subscription cursor duration of fetch",
+            exponential_buckets(1.0, 5.0, 11).unwrap(),
+        );
+        let subscription_cursor_fetch_duration =
+            register_histogram_vec_with_registry!(opts, &["subscription_name"], registry).unwrap();
+        Self {
+            _cursor_metrics_collector: CursorMetricsCollector::new(session_map, registry),
+            subscription_cursor_error_count,
+            subscription_cursor_query_duration,
+            subscription_cursor_declare_duration,
+            subscription_cursor_fetch_duration,
+        }
+    }
+}
+
+pub struct PeriodicCursorMetrics {
+    pub subsription_cursor_nums: i64,
+    pub invalid_subsription_cursor_nums: i64,
+    pub subscription_cursor_last_fetch_duration: HashMap<String, f64>,
+}
+
+#[derive(Clone)]
+struct CursorMetricsCollector {
+    join_handle: Arc<JoinHandle<()>>,
+}
+impl CursorMetricsCollector {
+    fn new(session_map: SessionMapRef, registry: &Registry) -> Self {
+        let subsription_cursor_nums = register_int_gauge_with_registry!(
+            "subsription_cursor_nums",
+            "The number of subscription cursor",
+            registry
+        )
+        .unwrap();
+        let invalid_subsription_cursor_nums = register_int_gauge_with_registry!(
+            "invalid_subsription_cursor_nums",
+            "The number of invalid subscription cursor",
+            registry
+        )
+        .unwrap();
+
+        let opts = histogram_opts!(
+            "subscription_cursor_last_fetch_duration",
+            "Since the last fetch, the time up to now",
+            exponential_buckets(1.0, 5.0, 11).unwrap(),
+        );
+        let subscription_cursor_last_fetch_duration =
+            register_histogram_vec_with_registry!(opts, &["subscription_name"], registry).unwrap();
+
+        let join_handle = tokio::spawn(async move {
+            loop {
+                tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
```

Comment:
Please define a const for this magic number.

Reply:
Also, 1 sec seems too frequent for metrics collection, considering that the collection overhead is not negligible (there are lock acqusition). I suggest we change it to 60s and use `tokio::time::interval` for the implementation. Example: https://github.com/risingwavelabs/risingwave/blob/3e7d5c2ea2bc14158f6f8c8dfff45fcc530fb2d0/src/meta/src/rpc/metrics.rs#L996